community/consistency-check/src/main/java/org/neo4j/consistency/internal/SchemaIndexExtensionLoader.java

```diff
@@ -0,0 +1,81 @@
+
+import java.io.File;
+
+import org.neo4j.helpers.Service;
+import org.neo4j.index.internal.gbptree.CleanupJob;
+import org.neo4j.index.internal.gbptree.RecoveryCleanupWorkCollector;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.kernel.api.index.SchemaIndexProvider;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.extension.KernelExtensionFactory;
+import org.neo4j.kernel.extension.KernelExtensions;
+import org.neo4j.kernel.extension.UnsatisfiedDependencyStrategies;
+import org.neo4j.kernel.extension.dependency.AllByPrioritySelectionStrategy;
+import org.neo4j.kernel.impl.api.index.SchemaIndexProviderMap;
+import org.neo4j.kernel.impl.factory.DatabaseInfo;
+import org.neo4j.kernel.impl.logging.LogService;
+import org.neo4j.kernel.impl.spi.KernelContext;
+import org.neo4j.kernel.impl.spi.SimpleKernelContext;
+import org.neo4j.kernel.impl.transaction.state.DefaultSchemaIndexProviderMap;
+import org.neo4j.kernel.impl.util.Dependencies;
+import org.neo4j.kernel.lifecycle.LifecycleAdapter;
+
+public class SchemaIndexExtensionLoader
+{
+    public static SchemaIndexProviderMap loadSchemaIndexProviders( KernelExtensions extensions )
+    {
+        AllByPrioritySelectionStrategy<SchemaIndexProvider> indexProviderSelection = new AllByPrioritySelectionStrategy<>();
+        SchemaIndexProvider defaultIndexProvider =
+                extensions.resolveDependency( SchemaIndexProvider.class, indexProviderSelection );
+        return new DefaultSchemaIndexProviderMap( defaultIndexProvider,
+                indexProviderSelection.lowerPrioritizedCandidates() );
+    }
+
+    @SuppressWarnings( "unchecked" )
+    public static KernelExtensions instantiateKernelExtensions( File storeDir, FileSystemAbstraction fileSystem,
+            Config config, LogService logService, PageCache pageCache,
+            RecoveryCleanupWorkCollector recoveryCollector, DatabaseInfo databaseInfo )
+    {
+        Dependencies deps = new Dependencies();
+        deps.satisfyDependencies( fileSystem, config, logService, pageCache, recoveryCollector );
+        @SuppressWarnings( "rawtypes" )
+        Iterable kernelExtensions = Service.load( KernelExtensionFactory.class );
+        KernelContext kernelContext = new SimpleKernelContext( storeDir, databaseInfo, deps );
+        return new KernelExtensions( kernelContext, kernelExtensions, deps, UnsatisfiedDependencyStrategies.ignore() );
+    }
+
+    private static class RecoveryPreventingCollector extends LifecycleAdapter implements RecoveryCleanupWorkCollector
+    {
+        @Override
+        public void add( CleanupJob job )
+        {
+            if ( job.needed() )
+            {
+                throw new IllegalStateException( "Consistency checker should not do recovery" );
+            }
+        }
+    }
+
+    public static final RecoveryCleanupWorkCollector RECOVERY_PREVENTING_COLLECTOR = new RecoveryPreventingCollector();
```

Comment:
well if you will need in any other test ever RecoveryPreventingCollector I would say that SchemaIndexExtensionLoader would be the last place where i would expect to find it, even in separate class is better imho